specs/RuntimeSpec/Actions.md

```diff
@@ -32,35 +43,48 @@ _Requirements:_
 - sets Account `storage_usage` to `account_cost` (genesis config)
 - sets Account `storage_paid_at` to the current block height
 
-NOTE: for the all subsequent actions in the transaction the `signer_id` becomes `receiver_id` until [DeleteAccountAction](#DeleteAccountAction). It allows to execute actions on behalf of the just created account.
-
+**Errors**:
+- If the action tries to create a top level account whose length is no greater than 32 characters, and `predecessor_id` is not
+`registrar_account_id`, the following error will be returned
 ```rust
-pub struct CreateAccountAction {}
+/// A top-level account ID can only be created by registrar.
+CreateAccountOnlyByRegistrar {
+    account_id: AccountId,
+    registrar_account_id: AccountId,
+    predecessor_id: AccountId,
+}
 ```
 
-## DeployContractAction
-
-_Requirements:_
+- If the action tries to create an account that is neither a top-level account or a subaccount of `predecessor_id`,
+the following error will be returned
+```rust
+/// A newly created account must be under a namespace of the creator account
+CreateAccountNotAllowed { account_id: AccountId, predecessor_id: AccountId },
+```
 
-- _`tx.signer_id` to be equal to `receiver_id`_
-- _`tx.public_key` to be `AccessKeyPermission::FullAccess` for the `singer_id`_
+NOTE: for the all subsequent actions in the transaction the `signer_id` becomes `receiver_id` until [DeleteAccountAction](#DeleteAccountAction). It allows to execute actions on behalf of the just created account.
```

Comment:
Found this note. good. See comment below for `actor_id`

Instead of `signer_id` you should use `predecessor_id`. Actions can be created through a promise API without extra transaction. Since this section talks about Actions in general context and not only actions within a transaction, then we should use `predecessor_id` whenever possible and forget about `signer_id`